src/app/volumenes/[dateYear]/page.tsx

```diff
@@ -0,0 +1,48 @@
+import volumenes from "@/data/Volumenes";
+
+const Volume = ({ params }: any) => {
+  // Verificar si params está definido y tiene la propiedad dateYear
+  if (!params || !params.dateYear) {
+    // Mostrar todos los datos de volumenes
+    return (
+      <div className="flex min-h-screen flex-col items-center justify-between p-24">
```

Comment:
please fragment, your components, into small parts. 
you can divide this part of your code and create a component called `ItemVolumes`